internal/lock/sync.go

```diff
@@ -0,0 +1,92 @@
+package lock
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.jetpack.io/devbox/internal/cuecfg"
+)
+
+func SyncLockfiles() error {
+	latestPackages, err := latestPackages()
+	if err != nil {
+		return err
+	}
+
+	// Step 2: Update the devbox.lock files
+	return filepath.Walk(".", func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !fi.IsDir() && filepath.Base(path) == "devbox.lock" {
+			var lockFile File
+			if err := cuecfg.ParseFile(path, &lockFile); err != nil {
+				return err
+			}
+
+			changed := false
+			for key, latestPkg := range latestPackages {
+				if pkg, exists := lockFile.Packages[key]; exists {
+					if pkg.LastModified != latestPkg.LastModified {
+						lockFile.Packages[key].AllowInsecure = latestPkg.AllowInsecure
+						lockFile.Packages[key].LastModified = latestPkg.LastModified
+						// PluginVersion is intentionally omitted
+						lockFile.Packages[key].Resolved = latestPkg.Resolved
+						lockFile.Packages[key].Source = latestPkg.Source
+						lockFile.Packages[key].Version = latestPkg.Version
+						lockFile.Packages[key].Systems = latestPkg.Systems
+						changed = true
+					}
+				}
+			}
+
+			if changed {
+				if err = cuecfg.WriteFile(path, lockFile); err != nil {
+					return err
+				}
+				fmt.Printf("Updated: %s\n", path)
+			}
+		}
+		return nil
+	})
+}
+
+func latestPackages() (map[string]*Package, error) {
+	latestPackages := make(map[string]*Package)
+
+	err := filepath.Walk(".", func(path string, fi os.FileInfo, err error) error {
```

Comment:
Is it possible to have a separate function that simply returns all the `devbox.json` file paths, so that we only walk it once? That way if a `.gitignore` doesn't catch a giant directory such as `node_modules`, at least the delay only happens once 🤷‍♀️

Reply:
will do